controllers/registry_controller.go

```diff
@@ -71,9 +86,129 @@ func (r *registryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 		err = markAsFreeOnSuccess(r.client, r.watchNamespace, r.recorder, RegistryController, result.Requeue, err)
 	}()
 
+	// List all IDMS/ITMS resources
+	imageDigestMirrorSetList := &config.ImageDigestMirrorSetList{}
+	if err = r.client.List(ctx, imageDigestMirrorSetList); err != nil {
+		return ctrl.Result{}, fmt.Errorf("error getting IDMS list: %w", err)
+	}
+	imageTagMirrorSetList := &config.ImageTagMirrorSetList{}
+	if err = r.client.List(ctx, imageTagMirrorSetList); err != nil {
+		return ctrl.Result{}, fmt.Errorf("error getting ITMS list: %w", err)
+	}
+
+	mergedMirrorSets := collectMirrorSets(imageDigestMirrorSetList.Items, imageTagMirrorSetList.Items)
+
 	return ctrl.Result{}, nil
 }
 
+// collectMirrorSets extracts the contents of the given mirror sets and merges them into a single collection.
+// Each member of the collection represents the registry configuration for a specific source.
+func collectMirrorSets(idmsItems []config.ImageDigestMirrorSet, idtsItems []config.ImageTagMirrorSet) []mirrorSet {
```

Comment:
Can these helper methods be placed into another package say registry and be unit tested in that package?

Reply:
I was thinking something similar with something like
```
type RegistryConfig struct {
  mirrors [] mirrorset
}
func NewRegistryConfig([]imageDigestMirrorSetList, ImageTagMirrorSetList) (*RegistryConfig,err){
  //Populates and sorts mirrorset list
  ...
}
// with method:

// outputs a config file representation
func (r * RegistryConfig) ToFile() []byte {}


```

Reply:
Sure I will create a new `registries` package and top level struct

I don't think I can implement ToFile yet, will likely have to wait for the next story due to the [same problem as here](https://github.com/openshift/windows-machine-config-operator/pull/2140#discussion_r1567940984)